### PR TITLE
vo_gpu/hwdec: rename and introduce legacy names for some interops

### PR DESCRIPTION
**Pull request by philipl**
We've had some annoying names for interops, which we can't simply rename because that would break config files and command lines. So we need to put a little more effort in and add a concept of legacy names that allow us to continue loading them, but with a warning.

The three I'm renaming here are:
* vaapi-egl -> vaapi (vaapi works with Vulkan too)
* drmprime-drm -> drmprime-overlay (actually describes what it does)
* cuda-nvdec -> cuda (cuda interop is not nvdec specific)

### CODE REVIEW

**Comment by Dudemanguy** [comment]
I'm in favor of the idea. `vaapi-egl`'s name is particularly extremely misleading.

File: video/out/gpu/hwdec.c [hunk 1]
```diff
@@ -275,6 +277,12 @@ void ra_hwdec_ctx_init(struct ra_hwdec_ctx *ctx, struct mp_hwdec_devices *devs,
             if (strcmp(type, drv->name) == 0) {
                 load_add_hwdec(ctx, devs, drv, false);
                 break;
+            } else if (strcmp(type, drv->legacy_name) == 0) {
+                MP_WARN(ctx, "gpu-hwdec-interop loaded by legacy name '%s'. "
+                             "Please change to '%s' in your config file or command line.\n",
```
Comment (Dudemanguy) [inline video/out/gpu/hwdec.c:282]:
```suggestion
                MP_WARN(ctx, "gpu-hwdec-interop was selected with the legacy name, '%s'. "
                             "Please change it to '%s' in your config file or command line.\n",
```

I dunno sounds better imo.

**Comment by Dudemanguy** [comment]
I see no reason to not also change `cuda-nvdec` while you're at it.

**Comment by sfan5** [comment]
To be clear this is about `--gpu-hwdec-interop` and not the names passed to `--hwdec`, right?
The docs says the following so all this work isn't strictly necessary:
> This option is for troubleshooting hwdec interop issues. Since it's a debugging option, its semantics may change at any time.

I also suspect that effectively nobody has an option in their config to force a specific interop.

**Comment by philipl** [comment]
That's true, but I didn't feel comfortable just making the breaking change. We can be more aggressive about dropping it though. Ship it in 0.35 and then drop right after.

**Comment by sfan5** [comment]
Fine by me.

File: video/out/gpu/hwdec.c [hunk 2]
```diff
@@ -279,6 +281,12 @@ void ra_hwdec_ctx_init(struct ra_hwdec_ctx *ctx, struct mp_hwdec_devices *devs,
             if (strcmp(type, drv->name) == 0) {
                 load_add_hwdec(ctx, devs, drv, false);
                 break;
+            } else if (strcmp(type, drv->legacy_name) == 0) {
+                MP_WARN(ctx, "gpu-hwdec-interop was selected with the legacy name '%s'. "
+                             "Please it change to '%s' in your config file or command line.\n",
```
Comment (Dudemanguy) [inline video/out/gpu/hwdec.c:286]: Accidentally put `it change` here.

Comment (sfan5) [inline video/out/gpu/hwdec.c:286]:
```suggestion
                             "Please change it to '%s' in your config file or command line.\n",
```

Comment (philipl) [reply video/out/gpu/hwdec.c]: Whoops. I made the fix but forgot to squash it.

**Review by Dudemanguy — APPROVED** [review]

**Pull request merged**